src/server/game/Entities/Unit/Unit.cpp

```diff
@@ -5531,12 +5531,30 @@ void Unit::SendAttackStateUpdate(uint32 HitInfo, Unit* target, uint8 /*SwingType
     SendAttackStateUpdate(&dmgInfo);
 }
 
-void Unit::SetPowerType(Powers new_powertype, bool sendUpdate/* = true*/)
+void Unit::InitPowerType()
```

Comment:
Why is this in Unit if only creatures use this? Also about `UnitsUseDefaultPowerOnInit` - `Unit` means Creature, not our Unit class

Reply:
My idea was to use this function also for players (flag SetToMaxOnInitialLogIn)

About UnitsUseDefaultPowerOnInit, there are some PowerTypes with this flag but without the flag IsUsedByNPCs